pkg/gen/nodejs-templates/helm.ts.mustache

```diff
@@ -63,63 +63,76 @@ export namespace v2 {
-            const overrides = tmp.fileSync({ postfix: ".yaml" });
-            const chartDir = tmp.dirSync({ unsafeCleanup: true });
-
-            try {
-                let chart: string;
-                let defaultValues: string;
-                if (isChartOpts(config)) {
-                    // Fetch chart.
-                    fetch(`${config.repo}/${config.chart}`, {
-                        destination: chartDir.name,
-                        version: config.version
-                    });
-                    chart = path.quotePath(nodepath.join(chartDir.name, config.chart));
-                    defaultValues = path.quotePath(
-                        nodepath.join(chartDir.name, config.chart, "values.yaml")
-                    );
-                } else {
-                    chart = path.quotePath(config.path);
-                    defaultValues = path.quotePath(nodepath.join(chart, "values.yaml"));
-                }
-
-                // Write overrides file.
-                const data = JSON.stringify(config.values || {}, undefined, "  ");
-                fs.writeFileSync(overrides.name, data);
-
-                // Does not require Tiller. From the `helm template` documentation:
-                //
-                // >  Render chart templates locally and display the output.
-                // >
-                // > This does not require Tiller. However, any values that would normally be
-                // > looked up or retrieved in-cluster will be faked locally. Additionally, none
-                // > of the server-side testing of chart validity (e.g. whether an API is supported)
-                // > is done.
-                const release = shell.quote([releaseName]);
-                const values = path.quotePath(overrides.name);
-                const namespaceArg = config.namespace
-                    ? `--namespace ${shell.quote([config.namespace])}`
-                    : "";
-                const yamlStream = execSync(
-                    `helm template ${chart} --name ${release} --values ${defaultValues} --values ${values} ${namespaceArg}`
-                ).toString();
-                this.resources = this.parseTemplate(yamlStream, config.transformations);
-            } catch (e) {
-                // Shed stack trace, only emit the error.
-                throw new pulumi.RunError(e.toString());
-            } finally {
-                // Clean up temporary files and directories.
-                chartDir.removeCallback();
-                overrides.removeCallback();
-            }
+			const allConfig = pulumi.output(config);
+			const configDeps = Array.from(<Set<pulumi.Resource>>(<any>allConfig).resources());
+
+			(<any>allConfig).isKnown.then((isKnown: boolean) => {
+				if (!isKnown) {
+					// Note that this can only happen during a preview.
+					pulumi.log.info("Note: some chart inputs are unknown; the chart's resources will not be displayed.");
```

Comment:
My only big UX feedback in this PR is:

* I'd really like this to be correlated to the actual chart resource. Right now this `info` is displayed under the stack.
* As an `info` it's kind of easy to miss because it appears in the diagnostics summary. It would be nice if this was integrated as a first-class citizen in the display, like:
  
  ```
  
       Type                                         Name                   Plan       Info
   +   pulumi:pulumi:Stack                          azure-mean-azure-mean  create     
   +   └─ kubernetes:helm.sh:Chart                  node                   create     [Can't preview] all chart values must be known ahead of time to generate an accurate preview
  ```

  If we're not willing to invest in that, maybe we should do a `warning`? I don't know. It is a warning in the sense you should tread carefully and you want to get users attention, but it's also _not_ in the sense that you can't really correct it.
* I (personally?) find the wording here a little strange. I don't think what I'm suggesting above is optimal, but I do think we should mention specifically that it's the _values_ passed to the chart that is causing this. Helm users will understand that better than "inputs".